### PR TITLE
Bluetooth: Audio: Update documentation for bt_audio_codec_qos

### PR DESCRIPTION
**Pull request by Thalley**
The struct and its fields were poorly documented.
The new documentation should she some light on when the fields can be modified.

### CODE REVIEW

File: include/zephyr/bluetooth/audio/audio.h [hunk 1]
```diff
@@ -702,60 +702,95 @@ enum {
...
-	/** @brief Maximum PDU size
-	 *
-	 *  Maximum size, in octets, of the payload from link layer to link
-	 *  layer.
+	/**
+	 * @brief Connected Isochronous Group (CIG) parameters
 	 *
-	 *  Value range @ref BT_ISO_PDU_MIN to @ref BT_ISO_PDU_MAX.
+	 * The following parameters in this struct affect the value sent to the controller via HCI
+	 * when creating the CIG. Once the group have been created with
```
Comment (kruithofa) [inline include/zephyr/bluetooth/audio/audio.h:721]:
```suggestion
	 * when creating the CIG. Once the group has been created with
```

File: include/zephyr/bluetooth/audio/audio.h [hunk 2]
```diff
@@ -702,60 +702,95 @@ enum {
...
+		 * Allowed values are @ref BT_AUDIO_CODEC_QOS_1M, @ref BT_AUDIO_CODEC_QOS_2M and
+		 * @ref BT_AUDIO_CODEC_QOS_CODED.
+		 */
+		uint8_t phy;
+
+		/**
+		 * @brief Retransmission Number
+		 *
+		 * This a recommendation to the controller, and the actual retransmission number
+		 * may be different that this.
```
Comment (kruithofa) [inline include/zephyr/bluetooth/audio/audio.h:740]:
```suggestion
		 * may be different from this.
```

**Review by kruithofa — CHANGES_REQUESTED** [review]
Two small grammatical things

**Review by kruithofa — DISMISSED** [review]

File: include/zephyr/bluetooth/audio/audio.h [hunk 3]
```diff
@@ -702,60 +702,95 @@ enum {
...
-	uint16_t latency;
-#endif /*  CONFIG_BT_BAP_BROADCAST_SOURCE || CONFIG_BT_BAP_UNICAST */
-
-	/** QoS Frame Interval */
-	uint32_t interval;
-
-	/** @brief QoS Presentation Delay in microseconds
+	 * This value can be changed up and until bt_bap_stream_qos() has been called.
+	 * Once a stream has been QoS configured, modifying this field does not modify the value.
+	 * It is however possible to modify this value and call bt_bap_stream_qos() again to update
```
Comment (larsgk) [inline include/zephyr/bluetooth/audio/audio.h:710]:
```suggestion
	 * It is however possible to modify this field and call bt_bap_stream_qos() again to update
```

File: include/zephyr/bluetooth/audio/audio.h [hunk 4]
```diff
@@ -702,60 +702,95 @@ enum {
...
-#if defined(CONFIG_BT_ISO_TEST_PARAMS)
-	/** @brief Maximum PDU size
-	 *
-	 *  Maximum size, in octets, of the payload from link layer to link
-	 *  layer.
+	/**
+	 * @brief Connected Isochronous Group (CIG) parameters
 	 *
-	 *  Value range @ref BT_ISO_PDU_MIN to @ref BT_ISO_PDU_MAX.
+	 * The following parameters in this struct affect the value sent to the controller via HCI
```
Comment (larsgk) [inline include/zephyr/bluetooth/audio/audio.h:720]:
```suggestion
	 * The fields in this struct affect the value sent to the controller via HCI
```

**Review by larsgk — COMMENTED** [review]
just a few comments

**Review by larsgk — APPROVED** [review]

**Review by kruithofa — APPROVED** [review]

**Pull request merged**